tests/unit/test_zfsacl.py

```diff
@@ -0,0 +1,113 @@
+import os
+import pwd
+import pytest
+import shutil
+import truenas_pylibzfs
+import zfsacl
+from contextlib import contextmanager
+from truenas_api_client import Client, ClientException
+
+
+@pytest.fixture(scope='module')
+def local_user():
+    with Client() as c:
+        usr = c.call('user.create', {
+            'username': 'test_local_user',
+            'full_name': 'test_local_user',
+            'random_password': True,
+            'group_create': True
+        })
+        try:
+            yield usr
+
+        finally:
+            c.call('user.delete', usr['id'])
+
+
+@pytest.fixture(scope='module')
+def create_dataset():
+    """ create an nfsv4 dataset under /root """
+    with Client() as c:
+        mnt = c.call('filesystem.mount_info', [['mountpoint', '=', '/var']], {'get': True})
+
+    lz = truenas_pylibzfs.open_handle()
+    ds_name = f'{mnt["mount_source"]}/nfsv4'
+    lz.create_resource(name=ds_name, type=truenas_pylibzfs.ZFSType.ZFS_TYPE_FILESYSTEM, properties={
+        truenas_pylibzfs.ZFSProperty.ACLTYPE: 'nfsv4',
+        truenas_pylibzfs.ZFSProperty.ACLMODE: 'restricted',
+    })
+    rsrc = lz.open_resource(name=ds_name)
+    try:
+        rsrc.mount()
+        yield rsrc
+    finally:
+        rsrc.unmount()
+        lz.destroy_resource(name=ds_name)
+
+
+@pytest.fixture(scope='function')
+def testdir(create_dataset):
+    os.mkdir('/var/nfsv4/testdir')
+
+    try:
+        yield '/var/nfsv4/testdir'
+    finally:
+        shutil.rmtree('/var/nfsv4/testdir')
+
+
+@contextmanager
+def run_as_user(username: str):
+    """
+    Temporarily assume the identity of `username`
+    (uid, gid, and supplementary groups), then restore.
+
+    Must be run as root.
+    """
+
+    # Save original credentials
+    orig_euid = os.geteuid()
+    orig_egid = os.getegid()
+    orig_groups = os.getgroups()
+
+    pw = pwd.getpwnam(username)
+
+    try:
+        # Drop privileges (order matters)
+        os.setgroups([])
+        os.initgroups(pw.pw_name, pw.pw_gid)
+        os.setegid(pw.pw_gid)
+        os.seteuid(pw.pw_uid)
+
+        yield
+
+    finally:
+        # Restore original credentials (reverse order)
+        os.seteuid(orig_euid)
+        os.setegid(orig_egid)
+        os.setgroups(orig_groups)
+
+
+def test_chown(local_user, testdir):
+    """ basic test that with trivial ACL owner cannot by default chown """
+    os.chown(testdir, local_user['uid'], -1)
```

Comment:
Presumably, -1 is supposed to be the GID for nobody (65534).  Maybe add a comment to that effect.

Reply:
Mark pointed out
> If the owner or group is specified as -1, then that ID is not changed.